### PR TITLE
Add cancel a request functionality to decoupled references

### PR DESCRIPTION
**Pull request by george-bit**
## Context

As part of the decoupled references feature candidates will need the ability to cancel their reference requests.

## Changes proposed in this pull request

Minor tweaks to content as per labels and wording but otherwise the same functionality that exists within the current referees section.

## Guidance to review

Check against the card criteria with Feature Flag on and off in review. Let me know if test coverage is suffiecient

## Link to Trello card

https://trello.com/c/mAoR8HLs/2222-%F0%9F%92%94-dev-cancel-a-reference-request

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -115,6 +115,11 @@ en:
     referee_type: What kind of referee do you want to add?
     referee_destroy: Are you sure you want to delete this referee?
     referee_cancel: Are you sure you want to cancel this request for a reference?
+<<<<<<< HEAD
```
Comment (TobyRet) [inline config/locales/en.yml:118]: @george-bit - woops!

Comment (george-bit) [reply config/locales/en.yml]: removed marker 👍

**Comment by george-bit** [comment]
@davidgisbey after speaking with @TobyRet,  I've created a spec file to cover the new CancelReference service, noticed that this does not exist currently for the RequestReference service? Do we have a plan for handling deprecation of the old referees code once the feature flag is removed? Would be good to have a discussion.

File: spec/system/candidate_interface/decoupled_references/review_references_spec.rb [hunk 2]
```diff
@@ -115,6 +116,17 @@ def and_i_can_delete_a_reference
...
 
+  def and_i_can_cancel_a_sent_reference
+    within '#references_sent' do
+      click_link 'Cancel request'
+    end
+    click_button 'Yes I’m sure'
+
+    expect(page).to have_current_path candidate_interface_decoupled_references_review_path
+    expect(page).to have_content("Reference request cancelled for #{@requested_reference.name}")
+    expect(page).to have_content('Cancelled')
```
Comment (malcolmbaig) [inline spec/system/candidate_interface/decoupled_references/review_references_spec.rb:127]: Worth checking the referee received the `reference_cancelled_email`

File: spec/services/cancel_referee_spec.rb [hunk 3]
```diff
@@ -1,4 +1,5 @@
 require 'rails_helper'
+# TODO: Remove file with Decoupled references FeatureFlag as part of deprication
```
Comment (malcolmbaig) [inline spec/services/cancel_referee_spec.rb:2]: `deprecation`

Comment (george-bit) [reply spec/services/cancel_referee_spec.rb:2]: D'oh

File: config/locales/en.yml [hunk 4]
```diff
@@ -115,6 +115,8 @@ en:
     referee_type: What kind of referee do you want to add?
     referee_destroy: Are you sure you want to delete this referee?
     referee_cancel: Are you sure you want to cancel this request for a reference?
+    decoupled_reference_cancel: Are you sure you want to cancel this reference request?
```
Comment (malcolmbaig) [inline config/locales/en.yml:118]: Maybe call this `references_cancel_request` for consistency with the other recent additions. Means no need to revisit this when we do clean up further down the road.

File: config/locales/en.yml [hunk 5]
```diff
@@ -115,6 +115,8 @@ en:
...
+    reference_destroy: Are you sure you want to delete this reference request?
```
Comment (malcolmbaig) [inline config/locales/en.yml:119]: Stray addition? Not seeing this used anywhere.

**Review by malcolmbaig — CHANGES_REQUESTED** [review]
LGTM, just a few bits to revisit.

File: config/locales/en.yml (hunk 5 above)
Comment (george-bit) [reply config/locales/en.yml:119]: Hmm yeah I think it must be. Thought it was in place already but must've added it experimentally. Got rid of it.

**Comment by davidgisbey** [comment]
LGTM once Malcolms changes are made. In terms of getting rid of the old code, all the referees/additional referees code should just be able to be deleted. The card i'm working on at the moment has pulled up a bunch of work that needs to be completed beforehand, including getting all the specs passing with the decoupled flag on, and moving the old specs into a specs_to_be_deleted folder. 

The final step of that card will be deleting the feature flag and all the old referee/additional referee code.

File: spec/system/candidate_interface/decoupled_references/review_references_spec.rb (hunk 2 above)
Comment (george-bit) [reply spec/system/candidate_interface/decoupled_references/review_references_spec.rb:128]: Given it a go. Think it's okay, let me know if not!

**Review by malcolmbaig — APPROVED** [review]

**Pull request merged**